components/raftstore-v2/src/operation/command/mod.rs

```diff
@@ -147,19 +206,160 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
+    }
+
+    #[inline]
+    async fn apply_entry(&mut self, entry: &Entry) -> Result<RaftCmdResponse> {
+        match SimpleWriteDecoder::new(entry.get_data()) {
+            Ok(decoder) => {
+                util::compare_region_epoch(
+                    decoder.header().get_region_epoch(),
+                    self.region_state().get_region(),
+                    false,
+                    true,
+                    true,
+                )?;
+                let res = Ok(new_response(decoder.header()));
+                for req in decoder {
+                    match req {
+                        SimpleWrite::Put(put) => self.apply_put(put.cf, put.key, put.value)?,
+                        SimpleWrite::Delete(delete) => self.apply_delete(delete.cf, delete.key)?,
+                        SimpleWrite::DeleteRange(dr) => self.apply_delete_range(
+                            dr.cf,
+                            dr.start_key,
+                            dr.end_key,
+                            dr.notify_only,
+                        )?,
+                    }
+                }
+                res
+            }
+            Err(req) => {
+                util::check_region_epoch(&req, self.region_state().get_region(), true)?;
+                if req.has_admin_request() {
+                    // TODO: implement admin request.
+                } else {
+                    for r in req.get_requests() {
+                        match r.get_cmd_type() {
+                            // These three writes should all use the new codec. Keep them here for
+                            // backward compatibility.
+                            CmdType::Put => {
+                                let put = r.get_put();
+                                self.apply_put(put.get_cf(), put.get_key(), put.get_value())?;
+                            }
+                            CmdType::Delete => {
+                                let delete = r.get_delete();
+                                self.apply_delete(delete.get_cf(), delete.get_key())?;
+                            }
+                            CmdType::DeleteRange => {
+                                let dr = r.get_delete_range();
+                                self.apply_delete_range(
+                                    dr.get_cf(),
+                                    dr.get_start_key(),
+                                    dr.get_end_key(),
+                                    dr.get_notify_only(),
+                                )?;
+                            }
+                            _ => unimplemented!(),
+                        }
+                    }
+                }
+                Ok(new_response(req.get_header()))
+            }
+        }
+    }
+
+    #[inline]
+    pub fn flush(&mut self) {
+        if let Some(wb) = self.write_batch_mut() && !wb.is_empty() {
+            let mut write_opt = WriteOptions::default();
+            write_opt.set_disable_wal(true);
```

Comment:
We don't have remove wal for multi-rocks yet, right?

Reply:
Yes, but it will. So implemented with the new design in mind.

Reply:
OK. I would suggest to add a TODO here but still keep it false. Until remove wal is implemented for multi-rocks then change it to true.

Reply:
That will not satisfy the performance requirement. Unless correctness test is involved, we don't need to set it to false.